### PR TITLE
Random set of patches on the compiler (WIP)

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

File: compiler/lib/javascript.ml [hunk 1]
```diff
@@ -163,10 +163,10 @@ and binop =
   | NotEq
   | EqEqEq
   | NotEqEq
-  | Lt
-  | Le
-  | Gt
-  | Ge
+  | Lt of bool
```
Comment (TyOverby) [inline compiler/lib/javascript.ml]: What is the bool here for?

File: compiler/lib/code.ml [hunk 2]
```diff
@@ -250,7 +250,6 @@ type prim =
   | Neq
   | Lt
   | Le
-  | Ult
```
[the remark this answers is not in the document]
Comment (hhugo) [reply compiler/lib/code.ml]: Usigned Lower Than

File: compiler/lib/generate.ml [hunk 3]
```diff
@@ -1199,6 +1192,7 @@ and translate_instr ctx expr_queue loc instr =
       let keep_name x =
         match Code.Var.get_name x with
         | None -> false
+        | Some "switcher" -> false
```
[the remark this answers is not in the document]
Comment (hhugo) [reply compiler/lib/generate.ml]: `switcher` is a name given by the ocaml compiler to some ident when compiling pattern matching. See https://github.com/ocaml/ocaml/blob/0da925c68df24923d6129a6270b563b111bd366a/lambda/matching.ml#L2263

File: compiler/lib/javascript.ml (hunk 1 above)
Comment (hhugo) [reply compiler/lib/javascript.ml]: This is a hack I used to know if the operator is comparing ints or something else.
Knowing that we're are comparing ints (and in particular, not floats) allows us do the following rewrite
`! ( a < b)` => `a >= b` (which would be wrong if `a` or `b` was `NaN`).
Ideally, such optimization should be done at an earlier stage. If we decide to keep the current approach, we should probably use inline records to explain the meaning of the Boolean.

**Comment by hhugo** [comment]
@TyOverby, theses commits should be read separately.
I'll probably split the PR up in smaller chunks.

**Pull request closed without merging**